hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,548 @@
+The {{< regref "ERROR_STATUS" >}} register will continue to report all of these violations even if one of the corresponding bits in {{< regref "ERROR_ENABLE" >}} is zero.
+
+The {{< regref "ERROR_STATUS" >}} bit should be cleared *before* clearing the error interrupt in the {{< regref "INTR_STATE" >}} register.
+Failure do to so may result in a repeated interrupt.
+
+## Status Indicators
+
+*TODO:* TXDQ and RXDQ
+
+## Other Registers
+
+### SPI_HOST Enable
+
+The SPI_HOST state machine is disabled on reset.
+Before any commands are processed, the block must be enabled by writing one to the {{< regref "CONTROL.SPIEN" >}} register.
+Writing a zero to this register temporarily suspends any previously submitted transactions.
+If the block is re-enabled by writing a one to {{< regref "CONTROL.SPIEN" >}}, any previously executing commands will continue from wherever they left off.
+
+An unacknowledged error interrupt will also suspend the core state machine.
+
+### Component reset
+
+In addition to the global hardware reset, there are three software reset options.
+- The main control FSM can be held in reset by writing a one to {{< regref "CONTROL.RST_FSM" >}}.
+This can be used to cancel a previous command.
+- The TX FIFO can be cleared by writing a one to {{< regref "CONTROL.RST_TXFIFO">}}
+- The RXFIFO can be cleared by writing a one to {{< regref "CONTROL.RST_RXFIFO">}}
+
+## Block Diagram
+
+![](spi_host_block_diagram.svg)
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/spi_host/data/spi_host.hjson" >}}
+
+## Design Details
+
+### Component Overview.
+
+**TODO** High level introductory description of
+- Control FSM
+- SDQ Shift register
+- Byte Select
+- Byte Merge
+- TX FIFO and RXFIFO interfaces
+
+### Command and Config CDC
+
+
+Highlights for this unit:
+- New commands can always be written to {{< regref "COMMAND_0" >}} or {{< regref "CONFIG_0" >}}
+- It is an error however to write a one to {{< regref "COMMAND_0.GO" >}} unless {{< regref "STATUS.READY" >}} is one.
+- The {{< regref "COMMAND_0.GO" >}} bit triggers a four-phase synchronizer, which copies the relevant multi-registers and `cs_n` masks to `coreCmdConf`.
+- {{<regref "STATUS.READY" >}} is low while this synchronizer is in operation
+- The core FSM is responsible for issuing the `cc_ack` signal.
+- `coreCmdCnf` is only updated and acknowledged (using `cc_ack`) when the FSM is not busy.
+
+{{< wavejson >}}
+{ signal: [
+  {name: "clk", wave: "p..............................."},
+  {name: "READY",         wave: "1.0..........1|.0..|...........1"},
+  {name: "COMMAND, CONFIG", wave: "x3............|4...|............"},
+  {name: "GO.q && GO.qe", wave: "010...........|10..|............"},
+  {name: "ccReq",        wave: "01.....0......|1...|.....0......"},
+  {name: "ccReqSyncd",  wave: "0..1.....0....|..1.|.......0...."},
+  {name: "ccAck",        wave: "0...1.....0...|....|..1.....0..."},
```

Comment:
Nit: The text above talks about `cc_ack`, rather than `ccAck`. Does these names need converting to camel case in the diagram?

Reply:
Good catch.   Changed all these to lower_snake_case.  I always forget when to use which.